tenant/duplicate_reads.go

```diff
@@ -48,7 +48,7 @@ type tenantService struct {
 }
 
 // NewReadOnlyStore returns a Store that cannot update the underlying kv.Store.
-func NewReadOnlyStore(store kv.Store) (*Store, error) {
+func NewReadOnlyStore(store kv.Store) *Store {
```

Comment:
I think the duplicate read system is no longer in use. You could delete this whole file if you want